src/scripts/view/template.ts

```diff
@@ -0,0 +1,105 @@
+import { Product, Review } from '../model/interfaces';
+
+export const templates = {
+    productCard: (product: Product) => {
+        return `
+        <div class="product-card" tabindex="0" data-id=${product.id}>
+            <div class="product-image">
+                <img
+                    src="${product.thumbnail}"
+                    loading="lazy"
+                    alt="${product.title}"
+                />
+            </div>
+            <div class="product-info">
+                <p>${product.title}</p>
+                <div class="price-row">
+                    <span class="price">$${product.price.toFixed(2)}</span>
+                    <span class="discount-badge">${product.discountPercentage}% off</span>
+                </div>
+                <p class="category">${product.category}</p>
+                <div class="rating">
+                    ${'★'.repeat(Math.floor(product.rating))}
+                    <span>(${product.rating})</span>
+                </div>
+                <p class="stock">${product.stock} in stock</p>
+            </div>
+            <div class="card-buttons">
+                <button class="delete-details-btn" aria-label="Delete product" id="${product.id}">
+                    Delete
+                </button>
+                <button class="edit-details-btn" aria-label="Edit product id="${product.id}">Edit</button>
+                <button class="view-details-btn" aria-label="View Details id="${product.id}">
+                    view details
+                </button>
+            </div>
+        </div>
+        `;
+    },
+
+    productDetails: (product: Product) => `
+        <div>
+            <div class="detail-header">
+                <div class="product-detail-image" >
+                    <img src="${product.thumbnail}" alt="${product.title}">
+                </div>
+                <div class="product-dis">
+                    <p class="product-title">${product.title}</p>
+                    <p class="product-brand">by ${product.brand}</p>
+                    <div class="product-price-des">
+                        <span class="product-detail-price">$${product.price}</span>
```

Comment:
We can also add it here as you added in `productCard` to ensure consistency ?

```suggestion
                        <span class="product-detail-price">$${product.price.toFixed(2)}</span>
```

Reply:
Added `toFix` (https://github.com/krushna-simform/product-management-system/pull/3/commits/b836ad679c5079aa203f2250f83158297ed3f874)